From d597c842991340ce2ccbb112715ec0f0eb3d21c3 Mon Sep 17 00:00:00 2001
From: Alexys Jacob <ultrabug@gentoo.org>
Date: Tue, 9 May 2017 09:18:32 +0200
Subject: [PATCH] fix perftune.py IndexError when NIC uses less IRQs than
 requested.

Signed-off-by: Alexys Jacob <ultrabug@gentoo.org>
---
 seastar/scripts/perftune.py | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/scripts/perftune.py b/scripts/perftune.py
index 6748086..63fb8fd 100755
--- a/seastar/scripts/perftune.py
+++ b/seastar/scripts/perftune.py
@@ -476,7 +476,8 @@ class NetPerfTuner(PerfTunerBase):
         Otherwise, we will use only IRQs which names fit one of the patterns above.
         """
         irqs2procline = get_irqs2procline_map()
-        all_irqs = learn_all_irqs_one("/sys/class/net/{}/device".format(iface), irqs2procline, iface)
+        # filter 'all_irqs' to only reference valid keys from 'irqs2procline' and avoid an IndexError on the 'irqs' search below
+        all_irqs = set(learn_all_irqs_one("/sys/class/net/{}/device".format(iface), irqs2procline, iface)).intersection(irqs2procline.keys())
         fp_irqs_re = re.compile("\-TxRx\-|\-fp\-|\-Tx\-Rx\-")
         irqs = list(filter(lambda irq : fp_irqs_re.search(irqs2procline[irq]), all_irqs))
         if len(irqs) > 0:
-- 
2.12.2

