From 9e89a00568ac56554922243d85b6b210f3ec841b Mon Sep 17 00:00:00 2001
From: Ultrabug <ultrabug@gentoo.org>
Date: Wed, 3 May 2017 17:21:37 +0200
Subject: [PATCH 03/13] coredump setup: add support for gentoo linux

---
 dist/common/scripts/scylla_coredump_setup | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/dist/common/scripts/scylla_coredump_setup b/dist/common/scripts/scylla_coredump_setup
index e4b50ba0..7fea51db 100755
--- a/dist/common/scripts/scylla_coredump_setup
+++ b/dist/common/scripts/scylla_coredump_setup
@@ -34,6 +34,8 @@ if [ "$ID" = "ubuntu" ] && [ "$VERSION_ID" != "16.04" ]; then
     sysctl -p /etc/sysctl.d/99-scylla.conf
 elif [ "$ID" = "debian" ]; then
     sysctl -p /etc/sysctl.d/99-scylla.conf
+elif [ "$ID" = "gentoo" ]; then
+    sysctl -p /etc/sysctl.d/99-scylla.conf
 else
     if [ "$ID" = "ubuntu" ]; then
         apt-get install -y systemd-coredump
-- 
2.12.2

