From a198f2b1afbfa341eabfdbaadcbea1111c716930 Mon Sep 17 00:00:00 2001
From: Alexys Jacob <ultrabug@gentoo.org>
Date: Wed, 17 May 2017 18:05:24 +0200
Subject: [PATCH 4/9] cpuscaling setup: add support for gentoo linux

Signed-off-by: Alexys Jacob <ultrabug@gentoo.org>
---
 dist/common/scripts/scylla_cpuscaling_setup | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/dist/common/scripts/scylla_cpuscaling_setup b/dist/common/scripts/scylla_cpuscaling_setup
index 5d02ae51..1089a77a 100755
--- a/dist/common/scripts/scylla_cpuscaling_setup
+++ b/dist/common/scripts/scylla_cpuscaling_setup
@@ -12,6 +12,16 @@ if is_debian_variant; then
         systemctl enable cpufrequtils.service
     fi
     service cpufrequtils start
+elif is_gentoo_variant; then
+    emerge -uq sys-power/cpupower
+    sed -e 's/--governor ondemand/--governor performance/g' -i /etc/conf.d/cpupower
+    if is_systemd; then
+        systemctl enable cpupower-frequency-set.service
+        systemctl start cpupower-frequency-set.service
+    else
+        rc-update add cpupower default
+        service cpupower start
+    fi
 else
     yum install -y cpupowerutils || true
     echo 'CPUPOWER_START_OPTS="frequency-set -g performance"' > /etc/sysconfig/cpupower
-- 
2.12.2

