From b78d665446c8b6d9a0893b3d18d419bbe17d702e Mon Sep 17 00:00:00 2001
From: Ultrabug <ultrabug@gentoo.org>
Date: Wed, 3 May 2017 17:22:57 +0200
Subject: [PATCH 07/13] raid setup: add support for gentoo linux

---
 dist/common/scripts/scylla_raid_setup | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/dist/common/scripts/scylla_raid_setup b/dist/common/scripts/scylla_raid_setup
index 8bb42cfe..36664b27 100755
--- a/dist/common/scripts/scylla_raid_setup
+++ b/dist/common/scripts/scylla_raid_setup
@@ -88,6 +88,8 @@ fi
 
 if is_debian_variant; then
     env DEBIAN_FRONTEND=noninteractive apt-get -y install mdadm xfsprogs
+elif is_gentoo_variant; then
+    emerge -uq sys-fs/mdadm sys-fs/xfsprogs
 else
     yum -y install mdadm xfsprogs
 fi
-- 
2.12.2

