From 3c058e8cc68a7560529438480f331a855fb4a3f3 Mon Sep 17 00:00:00 2001
From: Ultrabug <ultrabug@gentoo.org>
Date: Wed, 3 May 2017 17:24:54 +0200
Subject: [PATCH 09/13] scylla_setup: add gentoo linux installation detection

---
 dist/common/scripts/scylla_setup | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/dist/common/scripts/scylla_setup b/dist/common/scripts/scylla_setup
index 7d1de81e..17bbef84 100755
--- a/dist/common/scripts/scylla_setup
+++ b/dist/common/scripts/scylla_setup
@@ -74,6 +74,8 @@ interactive_ask_service() {
 verify_package() {
     if is_debian_variant; then
         dpkg -s $1 > /dev/null 2>&1 &&:
+    elif is_gentoo_variant; then
+        grep -q 'dev-db/scylla' /var/lib/portage/world
     else
         rpm -q $1 > /dev/null 2>&1 &&:
     fi
-- 
2.12.2

