From 16a0f0de4b167a76ea7f3630148ce32d80c15fa9 Mon Sep 17 00:00:00 2001
From: Ultrabug <ultrabug@gentoo.org>
Date: Wed, 3 May 2017 17:29:34 +0200
Subject: [PATCH 13/13] scylla_setup: fix typo on cpu scaling messages

---
 dist/common/scripts/scylla_setup | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/dist/common/scripts/scylla_setup b/dist/common/scripts/scylla_setup
index a5fe56cd..e14a9010 100755
--- a/dist/common/scripts/scylla_setup
+++ b/dist/common/scripts/scylla_setup
@@ -465,11 +465,11 @@ if [ $DEV_MODE -eq 1 ]; then
 fi
 
 if [ $INTERACTIVE -eq 1 ]; then
-    interactive_ask_service "Do you want to setup CPU scalling governor?" "Answer yes to  set CPU scalling governor to performance at boot time. Answer no to do nothing." "yes" &&:
+    interactive_ask_service "Do you want to setup CPU scaling governor?" "Answer yes to  set CPU scaling governor to performance at boot time. Answer no to do nothing." "yes" &&:
     CPUSCALING_SETUP=$?
 fi
 if [ $CPUSCALING_SETUP -eq 1 ]; then
-    run_setup_script "CPU Scalling" /usr/lib/scylla/scylla_cpuscaling_setup
+    run_setup_script "CPU Scaling" /usr/lib/scylla/scylla_cpuscaling_setup
 fi
 
 if is_redhat_variant; then
-- 
2.12.2

