diff --git a/gc_clock.hh b/gc_clock.hh
index ac3cbeb95..0e2ef5548 100644
--- a/gc_clock.hh
+++ b/gc_clock.hh
@@ -32,7 +32,7 @@
 class gc_clock final {
 public:
     using base = seastar::lowres_system_clock;
-    using rep = int64_t;
+    using rep = int32_t;
     using period = std::ratio<1, 1>; // seconds
     using duration = std::chrono::duration<rep, period>;
     using time_point = std::chrono::time_point<gc_clock, duration>;
diff --git a/sstables/mc/writer.cc b/sstables/mc/writer.cc
index d36789899..717766d8e 100644
--- a/sstables/mc/writer.cc
+++ b/sstables/mc/writer.cc
@@ -335,7 +335,7 @@ void write_delta_ttl(W& out, gc_clock::duration ttl, const encoding_stats& enc_s
 template <typename W>
 GCC6_CONCEPT(requires Writer<W>())
 void write_delta_local_deletion_time(W& out, int64_t local_deletion_time, const encoding_stats& enc_stats) {
-    write_unsigned_delta_vint(out, local_deletion_time, enc_stats.min_local_deletion_time.time_since_epoch().count());
+    write_unsigned_delta_vint(out, local_deletion_time, (int64_t)enc_stats.min_local_deletion_time.time_since_epoch().count());
 }
 
 template <typename W>
